### PR TITLE
fix(): look for slice id instead of slice name when creating or updat…

### PR DESCRIPTION
**Pull request by chengoconnell**
…ing slice

## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/slice-machine/server/src/api/services/sliceService.ts [hunk 1]
```diff
@@ -51,7 +49,7 @@ export async function uploadScreenshots(
 
     if (!!screenshot) {
       const { err, s3ImageUrl }: { err?: ApiError; s3ImageUrl?: string } =
-        await upload(env, sliceName, variationId, screenshot.path);
+        await upload(env, sliceModel, variationId, screenshot.path);
```
Comment (sintinilorenzo) [inline packages/slice-machine/server/src/api/services/sliceService.ts]: why are you passing the whole model down? Think it's better if you just pass down the slice ID if that's all you need

File: packages/slice-machine/server/src/api/slices/push.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ export async function pushSlice(
 
   try {
     const smModel: SliceSM = IO.Slice.readSlice(modelPath);
-    const { err } = await purge(env, slices, sliceName);
+    const { err } = await purge(env, slices, smModel);
```
Comment (sintinilorenzo) [inline packages/slice-machine/server/src/api/slices/push.ts]: same here "why are you passing the whole model down? Think it's better if you just pass down the slice ID if that's all you need"

File: packages/init/src/steps/starters/s3.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ async function updateVariationWithScreenshot(
   client: InitClient,
   acl: Acl,
   screenshotPaths: ComponentInfo["screenshotPaths"],
-  sliceName: SliceSM["id"],
+  sliceId: SliceSM["id"],
```
Comment (Heyrows) [inline packages/init/src/steps/starters/s3.ts:15]: this is clearly much more coherent 😂

File: packages/slice-machine/server/src/api/services/sliceService.ts [hunk 4]
```diff
@@ -14,13 +13,12 @@ import {
...
 export const createOrUpdate = async (
   slices: ReadonlyArray<SliceSM>,
-  sliceName: string,
   model: SliceSM,
   client: Client
 ) => {
   const prismicModel = Slices.fromSM(model);
 
-  if (slices.find((e) => e.id === snakelize(sliceName))) {
+  if (slices.find((e) => e.id === model.id)) {
```
Comment (Heyrows) [inline packages/slice-machine/server/src/api/services/sliceService.ts:21]: so we had everything we needed in the end 😂  nice 👍

File: packages/slice-machine/server/src/api/services/sliceService.ts (hunk 1 above)
Comment (Heyrows) [reply packages/slice-machine/server/src/api/services/sliceService.ts]: That would make the tests easier as well.

**Review by Heyrows — APPROVED** [review]
Beside the point raised by Lorenzo, the PR seems fine.

By changing the Purge function, what are the impact on the existing screenshots for the different scenarios ?

**Comment by chengoconnell** [comment]
> 

I checked with Mehdi on S3 and the old screenshots get deleted when we push a renamed slice to prismic! :)

File: packages/slice-machine/server/src/api/services/sliceService.ts (hunk 4 above)
Comment (chengoconnell) [reply packages/slice-machine/server/src/api/services/sliceService.ts:21]: thankfully yes! :)

**Review by sintinilorenzo — APPROVED** [review]

**Pull request merged**